pkg/scaffold/v1/controller/controller.go

```diff
@@ -199,8 +191,8 @@ type Reconcile{{ .Resource.Kind }} struct {
 // +kubebuilder:rbac:groups=apps,resources=deployments,verbs=get;list;watch;create;update;patch;delete
 // +kubebuilder:rbac:groups=apps,resources=deployments/status,verbs=get;update;patch
 {{ end -}}
-// +kubebuilder:rbac:groups={{.GroupDomain}},resources={{ .Plural }},verbs=get;list;watch;create;update;patch;delete
-// +kubebuilder:rbac:groups={{.GroupDomain}},resources={{ .Plural }}/status,verbs=get;update;patch
+// +kubebuilder:rbac:groups={{ .GroupDomain }},resources={{ .Resource.Plural }},verbs=get;list;watch;create;update;patch;delete
+// +kubebuilder:rbac:groups={{ .GroupDomain }},resources={{ .Resource.Plural }}/status,verbs=get;update;patch
```

Comment:
```suggestion
// +kubebuilder:rbac:groups={{ .GroupDomain }},resources={{ .Resource.Resource }}/status,verbs=get;update;patch
```

Reply:
As I said, `Resource.Resource` was being used for two different things, as its value could be changed with a flag for certain commands. Here we do not want the user-specific resource in case he provides it but the plural.

Reply:
Please, see that it has been used in the scaffolds implementations already instead of the plural as point out in the other comments.

Reply:
<img width="806" alt="Screenshot 2019-11-20 at 11 17 02" src="https://user-images.githubusercontent.com/7708031/69234583-661eb380-0b87-11ea-8e46-48e2d90fcc60.png">

Reply:
I insist, `resource.Resource` is not guaranteed to be the plural form, tahts only the default behavior. This file requires the plural form, not the one provided by the user. It was using `.Plural` for that. I just moved `.Plural` inside `Resource`, so that it is used as `.Resource.Plural`. In most of the cases `.Resource.Plural` will be the same as `.Resource.Resource` but not in all of them and we need the plural version here.